### PR TITLE
feat(components):Light version IconSwitch and DateTimePicker

### LINKED ISSUE #1581: [IconSwitch/DateTimePicker]: Support light version
### What package is this for?

- [X] React
- [ ] Angular

### Summary

- Support light version for components IconSwitch and DateTimePicker

**Additional context**
image

image

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

- ApplicationSuite release 8.2. (Oct. 2020)

### Available extra resources

Will be working on it.

### PR DESCRIPTION
**Pull request by cgirani**
Closes #1581 

**Summary**

- Adding css class for light version of component IconSwitch and DateTimePicker
- Adding id as a required key in preset object, since offset can be equal sometimes and more than one entry would appear as selected

eg.
image

**Change List (commits, features, bugs, etc)**

- feat(iconswitch): IconSwitch light version
- feat(datetimepicker): DateTimePicker light version
- fix(datetimepicker): use id as a preset selector

**Acceptance Test (how to verify the PR)**

- Make sure the components still works as expected.

### CODE REVIEW

File: src/components/DateTimePicker/_date-time-picker.scss [hunk 1]
```diff
@@ -181,6 +181,10 @@
       }
     }
   }
+
+  .#{$iot-prefix}--date-time-picker__box__light {
```
Comment (bjornalm) [inline src/components/DateTimePicker/_date-time-picker.scss:185]: I think light is a modifier so according to BEM we should probably write `.#{$iot-prefix}--date-time-picker__box--light`

File: src/components/DateTimePicker/DateTimePicker.jsx [hunk 2]
```diff
@@ -398,7 +409,7 @@ const DateTimePicker = ({
       value.kind = customRangeKind;
     } else {
       const preset = presets
-        .filter(p => p.offset === (clickedPreset ? clickedPreset.offset : selectedPreset))
+        .filter(p => p.id === (clickedPreset ? clickedPreset.id : selectedPreset))
```
Comment (bjornalm) [inline src/components/DateTimePicker/DateTimePicker.jsx:412]: I think this is a breaking change that affects the public API and thereby the versioning, @davidicus what is your take?

**Review by bjornalm — DISMISSED** [review]
Nice work! I just have two smaller change requests

File: src/components/DateTimePicker/DateTimePicker.jsx (hunk 2 above)
Comment (davidicus) [reply src/components/DateTimePicker/DateTimePicker.jsx:412]: yes. I think we can expand this function to check for `id` and if not present use the `offset`.

File: src/components/DateTimePicker/DateTimePicker.jsx [hunk 3]
```diff
@@ -466,7 +477,7 @@ const DateTimePicker = ({
   };
 
   const onPresetClick = preset => {
-    setSelectedPreset(preset.offset);
+    setSelectedPreset(preset.id);
```
Comment (davidicus) [inline src/components/DateTimePicker/DateTimePicker.jsx:480]: same comment on adding a fallback check for legacy.

File: src/components/DateTimePicker/DateTimePicker.jsx [hunk 4]
```diff
@@ -538,7 +549,7 @@ const DateTimePicker = ({
     // If value was changed reset when going back to Preset
     if (absoluteValue.startDate !== '' || relativeValue.lastNumber > 0) {
       if (selectedPreset) {
-        onPresetClick(presets.filter(x => x.offset === selectedPreset)[0]);
+        onPresetClick(presets.filter(x => x.id === selectedPreset)[0]);
```
Comment (davidicus) [inline src/components/DateTimePicker/DateTimePicker.jsx:552]: same need to support the `offset` check

File: src/components/DateTimePicker/DateTimePicker.jsx [hunk 5]
```diff
@@ -711,7 +726,7 @@ const DateTimePicker = ({
                         `${iotPrefix}--date-time-picker__listitem ${iotPrefix}--date-time-picker__listitem--preset`,
                         {
                           [`${iotPrefix}--date-time-picker__listitem--preset-selected`]:
-                            selectedPreset === preset.offset,
+                            selectedPreset === preset.id,
```
Comment (davidicus) [inline src/components/DateTimePicker/DateTimePicker.jsx:729]: same comment on `offset`

**Review by davidicus — DISMISSED** [review]

**Comment by tay1orjones** [comment]
@dianatran18 could you review and approve these two style variations? [`IconSwitch`](https://deploy-preview-1582--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-iconswitch--example-used-in-contentswitcher-light-version) and [`DateTimePicker`](https://deploy-preview-1582--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-experimental-datetime-picker--light-version). 

Are these available in sketch and documented on the internal website?

**Comment by dianatran18** [comment]
Both are good to go with the light variant 👍 

The IconSwitch is carried over in our new kit with the light variant option, but I haven't carried over the DateTimePicker yet. 

Although, I opened this issue before to make some stylistic updates to the IconSwitch (one being the light variant so that's good) on issue #1536. Could we get those updates in as well?

**Comment by tay1orjones** [comment]
@cgirani I think this is good to merge once @davidicus's comments are addressed. Let me know if you'd like me to take a look.

**Comment by bryancboyd** [comment]
Looks good 👍🏻

**Review by bryancboyd — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**